src/lib/file_system/file_system.ml

```diff
@@ -6,9 +7,14 @@ let remove_dir dir =
   let%bind _ = Process.run_exn ~prog:"rm" ~args:["-rf"; dir] () in
   Deferred.unit
 
-let try_finally ~(f : unit -> unit Deferred.t) ~finally =
-  let open Deferred.Or_error in
-  try_with f >>= (fun () -> ok_unit) |> ok_exn |> Deferred.bind ~f:finally
+let try_finally ~(f : unit -> 'a Deferred.t)
```

Comment:
this seems of general purpose, so maybe should be in a library other than `File_system`

Reply:
I agree, but there's no obvious place to put it, and it was in `File_system` already.